test/primitives.jl

```diff
@@ -22,6 +22,21 @@
     @test r(T(1.)) == P2(1,1)
     @test r(T(Inf)) == P2(Inf,Inf)
     @test_throws DomainError(T(-1), "r(t) is not defined for t < 0.") r(T(-1))
+    
+    r1 = Ray((0,0,0), (1,0,0.))
+    r2 = Ray((1,1,1), (1,2,1.))
+    @test r1 != r2
+  
+    r1 = Ray((0,0,0), (1,0,0.))
+    r2 = Ray((1,0,0), (-1,0,0.))
+    @test r1 != r2
+  
+    r1 = Ray((0,0,0), (1,0,0.))
+    @test r1 == r1
+    
+    r1 = Ray((0,0,0), (1,1,1.))
+    P1 = Point(3,3,3)
+    @test P1 ∈ r1
```

Comment:
We don't use capital letters for objects. Please rename P1 to p and r1 to r. Also, swap the lines because the function p \in r has arguments in the reverse order.